internal/route/router.go

```diff
@@ -106,7 +107,24 @@ func (r *SimpleRouter) Route(requestBody jsonrpc.RequestBody) (*jsonrpc.Response
 	}
 
 	zap.L().Debug("Routing request to upstream.", zap.String("upstreamID", id), zap.Any("request", requestBody))
+	metrics.UpstreamRPCRequestsTotal.WithLabelValues(id, configToRoute.HTTPURL).Inc()
 
+	body, response, err := r.routeToConfig(requestBody, &configToRoute)
+
+	if err != nil {
+		metrics.UpstreamRPCRequestErrorsTotal.WithLabelValues(id, configToRoute.HTTPURL).Inc()
+	} else if body.Error != nil {
+		zap.L().Warn("Encountered error in upstream JSONRPC response.", zap.Any("request", requestBody), zap.Any("error", body.Error))
+		metrics.UpstreamRPCRequestErrorsTotal.WithLabelValues(id, configToRoute.HTTPURL).Inc()
```

Comment:
What do you think about adding response code as a label?

Maybe `http-${statuscode}` and `jsonrpc-${statuscode}`?